builtin/providers/google/resource_storage_bucket.go

```diff
@@ -0,0 +1,149 @@
+package google
+
+import (
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/storage/v1"
+	"log"
+)
+
+func resourceStorageBucket() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceStorageBucketCreate,
+		Read:   resourceStorageBucketRead,
+		Update: resourceStorageBucketUpdate,
+		Delete: resourceStorageBucketDelete,
+
+		Schema: map[string]*schema.Schema{
+			"bucket": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"predefined-acl": &schema.Schema{
+				Type:     schema.TypeString,
+				Default:  "private",
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
```

Comment:
A discussion about ForceNew. From terraform (comments)[https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go]: ```
// If ForceNew is true, then a change in this resource necessitates
// the creation of a new resource.
```
Bucket location can't really be updated - if you want to change it, you are creating a new bucket. This is the same behaviour as when changing bucket name... On the other side, I don't see yet why force_destroy or predefined-acl would require ForceNew.